thirdparty/mcap/mcap/reader.inl

```diff
@@ -503,7 +511,11 @@ Status McapReader::readSummaryFromScan_(IReadable& reader) {
   typedReader.onAttachment = [&](const Attachment& attachment, ByteOffset fileOffset) {
     AttachmentIndex attachmentIndex{attachment, fileOffset};
     attachmentIndexes_.emplace(attachment.name, attachmentIndex);
-    attachments_.emplace(attachment.name, attachment);
+    // Copy buffer for storage as original is freed when leaving scope
+    Attachment attachment_copy = attachment;
+    attachment_copy.data = (std::byte*)std::malloc(attachment.dataSize);
+    std::memcpy((void*)attachment_copy.data, attachment.data, attachment.dataSize);
+    attachments_.emplace(attachment_copy.name, attachment_copy);
```

Comment:
Consider having `std::move(attachment_copy)` to avoid copying the attachment_copy unnecessarily when emplicing it in the attachment_.